meinberlin/apps/plans/models.py

```diff
@@ -208,6 +208,28 @@ def published_projects(self):
             is_archived=False,
         )
 
+    @cached_property
+    def published_projects_count(self) -> int:
+        try:
+            prefetch_enabled = (
+                self.projects.prefetch_cache_name in self._prefetched_objects_cache
+            )
+        except AttributeError:
```

Comment:
You mean `self._prefetched_objects_cache`? That's not a function, it's a dict. I've checked that it is working by fixing the value of `prefetch_enabled` and noting that the response time changes significantly. And I have checked that `self._prefetched_objects_cache` is a dict by debugging with a break point.

Reply:
to recap, if we don't want to use this hack, we can move the calculation with python to the serializer method, because we know it is only called from the view, and the view does a prefetch. @goapunk?

Reply:
I have added a function `is_prefetched(..)` to a new file `plans/helpers.py` to separate the way we count from testing the prefetch.

I guess if we move the code to the serializer we can silently assume it has been prefetched, but for clarity, I would leave the test in to be explicit about why we suddenly count in python. But either way works.

So two decisions pending: 1) move code to serializer? and 2) test whether projects has been prefetched?

Reply:
done, moved it to serializer